### ISSUE TITLE
Can Terraform be used to enable Fleet/Integrations Keep Policies Up to Date?

### ISSUE THREAD

**Issue by rheppe** [elastic/terraform-provider-elasticstack#672]
As the Fleet documentation mentions some integrations are installed (and maybe managed) by Fleet and some or all of them have an option in the 'Settings' tab of the integration to Keep integration up to date automatically.  Other integrations do not have this capability and would be updated manually when a new version is available.
https://www.elastic.co/guide/en/fleet/current/upgrade-integration.html#upgrade-integration-policies-automatically

Terraform providers offers an integration installation capability where an option for 'version' specifies a specific version.  This option is a type string.
https://registry.terraform.io/providers/elastic/elasticstack/latest/docs/resources/fleet_integration

Questions are:  1) Could an value of "latest" or some other string be used in place of a specific numbered version to 'enable' terraform to Keep the integration(s) up to date?  2)  Could the Fleet API with 'keepPoliciesUpToDate' option be used to set/enable all integrations across the board to be kept update to date automatically?  3)  Is it advisable to try this or would there be any known or unintended consequences that might occur thereby making it something not to do or attempt?

It seems that since some of the integrations have the keep up to date feature and some do not, that there is a reason that not all integrations already have this Fleet UI toggle button.

**Comment by breml** [comment]
It is a reasonable question to be asked, if it does make sense to manage the fleet integrations with Terraform. We do it for one of our setups since roughly 9 months and there are some challenges attached to it. The main reason we still do it is, that it helps us to keep everything under control between the 3 deployments (dev, test, prod).

What we find challenging is to perform updates of the fleet integrations. We normally execute the update manually on dev and then "backport" the changes to Terraform by analyzing the differences in the execution plan. The main challenge there is rooted in the fact, that some of the properties (for example `vars`, if I recall correctly) are marked as sensitive in Terraform and therefore Terraform refuses to show the differences even if the property might not contain any sensitive value for some of the fleet integrations (where others actually do store sensitive values in this property). Since analyzing the differences in the JSON output of the Terraform plan became very painful (especially, since the values are JSON in JSON), we even built our own tool to assist with this (https://github.com/breml/tfreveal).

The bottom line is, that with the use of fleet integrations one faces the challange, that there are two different systems (Terraform and the install/update mechanism of Fleet), which change the state and therefore there is always drift, that one needs to handle in Terraform. Sofar we made it work, but compared with other systems, that we manage with Terraform, this is for sure one of the more painful ones.
